js/whitebit.js

```diff
@@ -974,64 +974,65 @@ module.exports = class whitebit extends Exchange {
          * @param {object} params extra parameters specific to the whitebit api endpoint
          * @returns {object} an [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
          */
-        let method = undefined;
         await this.loadMarkets ();
         const market = this.market (symbol);
         const request = {
             'market': market['id'],
             'side': side,
             'amount': this.amountToPrecision (symbol, amount),
         };
-        const stopPrice = this.safeNumber2 (params, 'stopPrice', 'activationPrice');
-        if (stopPrice !== undefined) {
-            // it's a stop order
+        const isLimitOrder = (type === 'limit');
+        const isMarketOrder = (type === 'market');
+        const stopPrice = this.safeNumber2 (params, 'triggerPrice', 'stopPrice');
+        const isStopOrder = (stopPrice !== undefined);
+        const timeInForce = this.safeString (params, 'timeInForce');
+        const fok = (timeInForce === 'FOK');
+        if (timeInForce === 'IOC' || timeInForce === 'PO') {
+            throw new NotSupported (this.id + ' only supports FOK timeInForce');
+        }
+        if (fok) {
+            if (type !== 'market' || stopPrice !== undefined) {
+                throw new NotSupported (this.id + ' only supports FOK for regular market orders');
```

Comment:
Hmm this is a bit contradictory to the above statement because above we're saying `we only accept FOK` and here `only FOK for regular market orders` implicitly saying we accept other TIF values.

Can you show me in the docs where they define TIF rules pls?

Reply:
There is a separate endpoint which corresponds to `market` + `FOK` in ccxt terms

https://github.com/whitebit-exchange/api-docs/blob/main/docs/Private/http-trade-v4.md#create-stock-market-order

![image](https://user-images.githubusercontent.com/87606888/181816483-6b744dc1-729d-4cfe-b7a1-8ed414d886fa.png)

Reply:
![image](https://user-images.githubusercontent.com/87606888/181816575-2a949c4f-e134-49fe-a731-d0d54ee7743b.png)

Reply:
@DoctorSlimm thanks for the info, I resolved the other comment so we can discuss this in a single place.

So from my understanding, this exchange supports two TIF: GTC is the default TFI and additionally FOK for market orders only, right?  

So I would suggest changing it a bit to make it more clear, we could probably do something like this (It is almost the same as you have now). 

```Javacript
(...)
const timeInForce = this.safeString (params, 'timeInForce');
const isMarketOrder = (type === 'market');
const isLimitOrder = (type === 'limit');
(...)

if (timeInForce === 'FOK') {
    if (!isMarketOrder || isStopOrder) {
         throw new ...
    }
    // prepare request for stock market
} else {
   // handle regular GTC orders here (market, limit, stop, etc)
}
```
Do you agree or have a better suggestion?

Reply:
Sure that sounds good